roles/azure_manage_networking_stack/tasks/create.yml

```diff
@@ -1,8 +1,30 @@
 ---
-- name: Check that resource group exists
-  azure.azcollection.azure_rm_resourcegroup_info:
-    name: "{{ azure_manage_networking_stack_resource_group }}"
-  register: result
+- name: Check virtual network
+  ansible.builtin.fail:
+    msg: Azure virtual network must be defined as azure_manage_networking_stack_virtual_network
+  when: azure_manage_networking_stack_virtual_network is not defined
+
+- name: Check subnet
+  ansible.builtin.fail:
+    msg: Azure subnet must be defined as azure_manage_networking_stack_subnet
+  when: azure_manage_networking_stack_subnet is not defined
+
+- name: Check virtual network CIDR address prefixes
+  ansible.builtin.fail:
+    msg: Azure virtual network CIDR address prefixes must be defined as azure_manage_networking_stack_vnet_address_prefixes_cidr
+  when: azure_manage_networking_stack_vnet_address_prefixes_cidr is not defined
+
+- name: Check subnet CIDR address prefixes
+  ansible.builtin.fail:
+    msg: Azure subnet CIDR address prefixes must be defined as azure_manage_networking_stack_subnet_address_prefixes_cidr
+  when: azure_manage_networking_stack_subnet_address_prefixes_cidr is not defined
+
+- name: Check that Azure Region is known
+  ansible.builtin.fail:
+    msg: Azure region must be defined as azure_load_balancer_with_public_ip_region
+  when:
+    - azure_manage_networking_stack_region is not defined
+    - rg_info.resourcegroups | length == 0
```

Comment:
I think this condition is redundant. The region must be specified in any case

Reply:
Can you please explain this?
As far as I understand the region is used for the resource group creation, so its not needed if the resource group is already created

Reply:
region is also used to create virtual network

Reply:
I think the roles should automatically determine the region based on the resource group if the region is not defined.  This gives a sane default but still lets them override it.

Reply:
But this is optional in VirtualNetwork, it use the location of the ResourceGroup if it not given
Am I wrong?
This is the relevan module code:
https://github.com/ansible-collections/azure/blob/3342a8e29ce877fe7fb92b49affbf9b5204799a1/plugins/modules/azure_rm_virtualnetwork.py#L271

Reply:
As discussed, in case the region doesn't exist, it determined based on the resource group.
Just needed to add `default(omit)` for case it doesn't exist